maven_iuvs/echelle.py

```diff
@@ -2160,11 +2168,19 @@ def fit_H_and_D(pig, wavelengths, spec, light_fits, CLSF, unc=1, IPH_bounds=(Non
             # Algorithms such as the Powell method don't return inverse hessian; for Powell it's because it doesn't take any derivatives. 
             # We can estimate the Hessian using stattools per this link.
             # https://stackoverflow.com/questions/75988408/how-to-get-errors-from-solved-basin-hopping-results-using-powell-method-for-loc
-            hessian = approx_hess2(bestfit.x, negloglikelihood, args=objfn_args)
+            hessian = approx_hess2(bestfit.x, negloglikelihood, 
+                                   args=objfn_args)
 
             if fit_IPH_component:
                 fit_uncert = np.sqrt(np.diag(inv(hessian)))
-            else:
+                # Sometimes we get nans if the epsilon of approx_hess2 is chosen 
+                # automatically. Check and recalculate if need be.
+                if np.isnan(fit_uncert).any():
+                    new_hessian = approx_hess2(bestfit.x, negloglikelihood, 
+                                   epsilon=1e-2, # Seems to be the magic number...
```

Comment:
Based on the approx_hess2 documentation (available here: https://www.statsmodels.org/dev/generated/statsmodels.tools.numdiff.approx_hess2.html ), the default is to use

`epsilon = EPS**(1/3)*x`

That is, the epsilon used in the hessian computation is a fraction of the fitted parameter value. The default value for EPS is `np.finfo(float).eps`, which on my machine is ~2e-16, giving `epsilon ≈ 6e-6*x` .

Probably we should also scale epsilon by the best fit value, so I recommend trying `epsilon=1e-2*bestfit.x` .

Reply:
This is now solved with the jax implementation